src/pages/billing/NewBillingProjectWizard/GCPBillingProjectWizard/GCPBillingProjectWizard.ts

```diff
@@ -0,0 +1,118 @@
+import { CSSProperties, useEffect, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import colors from 'src/libs/colors'
+import { getLocalPref, setLocalPref } from 'src/libs/prefs'
+import { GoogleBillingAccount } from 'src/pages/billing/models/GoogleBillingAccount'
+import { StepWizard } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepWizard'
+
+import { AddTerraAsBillingAccountUserStep } from './AddTerraAsBillingAccountUserStep'
+import { BillingAccountAccessStep } from './BillingAccountAccessStep'
+import { ContactAccountAdminToAddUserStep } from './ContactAccountAdminToAddUserStep'
+import { CreateTerraProjectStep } from './CreateTerraProjectStep'
+import { GoToGCPConsoleStep } from './GoToGCPConsoleStep'
+
+
+interface GCPBillingProjectWizardProps {
+  billingAccounts: Record<string, GoogleBillingAccount>
+  onSuccess: (string) => void
+  // calls Auth.ensureBillingScope, then re-loads billing accounts from rawls
+  authorizeAndLoadAccounts: () => Promise<void>
+}
+
+export const GCPBillingProjectWizard = ({ onSuccess, billingAccounts, authorizeAndLoadAccounts }: GCPBillingProjectWizardProps) => {
+  const persistenceId = 'billing'
+  const [accessToBillingAccount, setAccessToBillingAccount] = useState<boolean>(() => getLocalPref(persistenceId)?.accessToBillingAccount)
+  const [accessToAddBillingAccountUser, setAccessToAddBillingAccountUser] = useState<boolean | undefined>(() => getLocalPref(persistenceId)?.accessToAddBillingAccountUser)
+  // if the user can't add a billing account user, has the user verified that terra was added
+  const [verifiedUsersAdded, setVerifiedUsersAdded] = useState<boolean | undefined>(() => getLocalPref(persistenceId)?.verifiedUsersAdded || false)
+  const [refreshed, setRefreshed] = useState<boolean>(false)
+  const [activeStep, setActiveStep] = useState<number>(() => getLocalPref(persistenceId)?.activeStep || 1)
+
+  useEffect(() => {
+    setLocalPref(persistenceId, { activeStep, accessToBillingAccount, accessToAddBillingAccountUser, verifiedUsersAdded })
+  }, [persistenceId, activeStep, accessToBillingAccount, accessToAddBillingAccountUser, verifiedUsersAdded])
+
+
+  return h(StepWizard, {
+    title: 'Link a Google Cloud billing account to Terra',
+    intro: `The linked billing account is required to cover all Google Cloud data storage, compute and egress costs incurred in a Terra workspace.
+        Cloud costs are billed directly from Google and passed through Terra billing projects with no markup.`,
+  }, [
+    h(GoToGCPConsoleStep, { isActive: activeStep === 1, stepFinished: () => setActiveStep(2) }),
+    h(BillingAccountAccessStep, {
+      accessToBillingAccount,
+      setAccessToBillingAccount: (access: boolean) => {
+        setAccessToBillingAccount(access)
+        setAccessToAddBillingAccountUser(undefined)
+        setVerifiedUsersAdded(undefined) // reset verification of step 3
+        setRefreshed(false) // this might or might not be necessary
```

Comment:
Is it necessary?

Reply:
I don't know… the comment was from @blakery .

Reply:
Sorry, I have to resolve this to be able to merge….